### PR TITLE
feat(dialog): add padding to default slot

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #9869

## Summary

- adds css var `--calcite-dialog-content-space`
- sets padding on dialog by default
- adds screenshot test

### CODE REVIEW

**Review by jcfranco — APPROVED** [review]

**Comment by jcfranco** [comment]
Should this use `feat` as the commit type? It would match the PR title and related issue label.

File: packages/calcite-components/src/components/dialog/dialog.scss [hunk 1]
```diff
@@ -6,6 +6,7 @@
  * @prop --calcite-dialog-scrim-background-color: Specifies the background color of the component's scrim.
  * @prop --calcite-dialog-size-x: Specifies the width of the component, using `px`, `em`, `rem`, `vw`, or `%`. Does not exceed the viewport's width - applies when `placement="cover"` is set.
  * @prop --calcite-dialog-size-y: Specifies the height of the component, using `px`, `em`, `rem`, `vh`, or `%`. Does not exceed the viewport's height - applies when `placement="cover"` is set.
+ * @prop --calcite-dialog-content-space: Specifies the padding of the component's content.
```
[the remark this answers is not in the document]
Comment (driskull) [reply packages/calcite-components/src/components/dialog/dialog.scss:9]: @macandcheese what padding value should be used?

[the remark this answers is not in the document]
Comment (driskull) [reply packages/calcite-components/src/components/dialog/dialog.scss:9]: Ok, I see the issue. Rebuilding stories.

File: packages/calcite-components/src/components/dialog/dialog.scss [hunk 2]
```diff
@@ -113,6 +114,10 @@ calcite-panel {
   --calcite-panel-header-border-block-end: var(--calcite-border-width-sm) solid var(--calcite-dialog-border-color);
 }
 
+.content {
+  padding: var(--calcite-dialog-content-space, var(--calcite-internal-dialog-padding));
```
[the remark this answers is not in the document]
Comment (geospatialem) [reply packages/calcite-components/src/components/dialog/dialog.scss]: @macandcheese @ashetland @SkyeSeitz LMK if we need a follow-up issue and what the priority for tackling is if this is something we'd like to pursue. 💪🏻

**Pull request merged**